parquet/src/file/page_index/range.rs

```diff
@@ -0,0 +1,461 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+use crate::errors::ParquetError;
+use parquet_format::PageLocation;
+use std::cmp::Ordering;
+use std::collections::VecDeque;
+use std::ops::RangeInclusive;
+
+type Range = RangeInclusive<usize>;
+
+pub trait RangeOps {
+    fn is_before(&self, other: &Range) -> bool;
+
+    fn is_after(&self, other: &Range) -> bool;
+
+    fn count(&self) -> usize;
+}
+
+impl RangeOps for Range {
+    fn is_before(&self, other: &Range) -> bool {
+        self.end() < other.start()
+    }
+
+    fn is_after(&self, other: &Range) -> bool {
+        self.start() > other.end()
+    }
+
+    fn count(&self) -> usize {
+        self.end() - self.start() + 1
```

Comment:
```suggestion
        self.end() + 1 - self.start()
```

I think otherwise this will potentially underflow - possibly worth a test

Reply:
👍  learned